release-task-helper.gradle

```diff
@@ -199,14 +226,57 @@ class ReleaseTask extends DefaultTask {
 
 }
 
+project.rootProject.ext.defaultAllowedLicenses = fileLocation("./data/default-allowed-licenses.json")
+project.rootProject.ext.allLicenses = fileLocation("./data/allow-all-licenses.json")
+project.rootProject.ext.licenseNormalizerBundle = fileLocation("./data/license-normalizer-bundle.json")
+
 class GoCDPlugin implements Plugin<Project> {
   void apply(Project project) {
+    project.configure(project) {
+      project.allprojects { apply plugin: com.github.jk1.license.LicenseReportPlugin }
+    }
+
     GoCDPluginExtension gocdPluginExtension = project.extensions.create('gocdPlugin', GoCDPluginExtension)
 
-    project.tasks.create('githubRelease', ReleaseTask) { ReleaseTask thisTask ->
-      thisTask.dependsOn project.allprojects*.tasks*.findByName('assemble').findAll { it != null }
+    project.afterEvaluate {
+      initializeLicenseReporting(gocdPluginExtension, project)
+
+      project.tasks.create('githubRelease', ReleaseTask) { ReleaseTask thisTask ->
+        thisTask.dependsOn project.allprojects*.tasks*.findByName('assemble').findAll { it != null }
+      }
+    }
+  }
+
+  private void initializeLicenseReporting(GoCDPluginExtension gocdPluginExtension, Project project) {
+    Project pluginProject = gocdPluginExtension.pluginProject
+    pluginProject.tasks.assemble.dependsOn("checkLicense")
+    pluginProject.extensions.licenseReport.outputDir = gocdPluginExtension.licenseReport.outputDir ?: "$project.rootDir/build/license"
+    pluginProject.extensions.licenseReport.projects = gocdPluginExtension.licenseReport.projects ?: project.allprojects
+    pluginProject.extensions.licenseReport.configurations = gocdPluginExtension.licenseReport.configurations ?: ["compile"]
+    pluginProject.extensions.licenseReport.excludeGroups = gocdPluginExtension.licenseReport.excludeGroups
+    pluginProject.extensions.licenseReport.excludeOwnGroup = gocdPluginExtension.licenseReport.excludeOwnGroup
+    pluginProject.extensions.licenseReport.excludes = gocdPluginExtension.licenseReport.excludes
+    pluginProject.extensions.licenseReport.renderers = [new TextReportRenderer()]
+    pluginProject.extensions.licenseReport.importers = gocdPluginExtension.licenseReport.importers
+    pluginProject.extensions.licenseReport.filters = new LicenseBundleNormalizer(bundlePath: project.rootProject.licenseNormalizerBundle)
+    println project.rootProject.licenseNormalizerBundle
+    if (gocdPluginExtension.licenseReport.validateDependenciesLicense) {
+      pluginProject.extensions.licenseReport.allowedLicensesFile = project.rootProject.defaultAllowedLicenses
+    } else {
+      pluginProject.extensions.licenseReport.allowedLicensesFile = project.rootProject.allLicenses
```

Comment:
This may not actually work when the script is applied from http. Try tweaking a plugin to `apply` from this branch?

Reply:
Also: I don't think we should allow whitelisting all licenses, not even accidentally.

Reply:
> This may not actually work when the script is applied from http. Try tweaking a plugin to apply from this branch?

Sure, I will try it from this branch